### PR TITLE
Publish container image and run tests yml files added

### PR DESCRIPTION
**Pull request by horozal**

### CODE REVIEW

File: .github/workflows/publish-container-image.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+      - name: Cache Maven Packages
+        uses: actions/cache@v3
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('./**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
+
+      - name: Build necessary modules
+        run: mvn install -pl context-monitoring -am
```
[the remark this answers is not in the document]
Comment (horozal) [reply .github/workflows/publish-container-image.yml]: Removed it.

File: .github/workflows/run-tests.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      uses: actions/cache@v3
+      with:
+        path: ~/.m2
+        key: ${{ runner.os }}-m2-${{ hashFiles('./**/pom.xml') }}
+        restore-keys: ${{ runner.os }}-m2
+
+    - name: Run Tests
+      run: mvn --file pom.xml --batch-mode test
+      env:
+        SMARTCLIDE_CONTEXT_GITLAB_API_TOKEN: ${{ secrets.GITLAB_DEV_SMARTCLIDE_EU_API_TOKEN }}
```
[the remark this answers is not in the document]
Comment (horozal) [reply .github/workflows/run-tests.yml]: Removed the environment definition.

File: .github/workflows/publish-container-image.yml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    env:
+      CONTAINER_REGISTRY_USERNAME: ${GITHUB_ACTOR}
+      CONTAINER_REGISTRY_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      SMARTCLIDE_CONTEXT_GITLAB_API_TOKEN: ${{ secrets.GITLAB_DEV_SMARTCLIDE_EU_API_TOKEN }}
```
[the remark this answers is not in the document]
Comment (horozal) [reply .github/workflows/publish-container-image.yml]: Removed the environment definition.

File: .github/workflows/publish-container-image.yml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+      - name: Get short commit SHA
+        if: ${{ github.event_name == 'push' }}
+        id: get_short_commit_SHA
+        run: echo "::set-output name=sha8::$(echo ${GITHUB_SHA} | cut -c1-8)"
+
+      - name: Build Image
+        if: ${{ github.event_name == 'push' }}
+        env:
+          IMAGE_TAG: ${{ steps.get_short_commit_SHA.outputs.sha8}}
+        run: mvn test jib:build -pl smartclide-monitoring -Djib.to.auth.username="${CONTAINER_REGISTRY_USERNAME}" -Djib.to.auth.password="${CONTAINER_REGISTRY_TOKEN}" -Djib.to.tags="${IMAGE_TAG}"
```
[the remark this answers is not in the document]
Comment (horozal) [reply .github/workflows/publish-container-image.yml]: Updated `pom.xml` accordingly. 
Removed `-pl smartclide-monitoring -Djib.to.auth.username="${CONTAINER_REGISTRY_USERNAME}" -Djib.to.auth.password="${CONTAINER_REGISTRY_TOKEN}"`

File: .github/workflows/publish-container-image.yml [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+      - name: Get tag name
+        if: ${{ github.event_name == 'release' }}
+        id: get_tag_name
+        run: echo "::set-output name=TAG_NAME::${{ github.event.release.tag_name }}"
+
+      - name: Build Image for Release
+        if: ${{ github.event_name == 'release' }}
+        env:
+          IMAGE_TAG: ${{ steps.get_tag_name.outputs.TAG_NAME }}
+        run: mvn test jib:build -pl smartclide-monitoring -Djib.to.auth.username="${CONTAINER_REGISTRY_USERNAME}" -Djib.to.auth.password="${CONTAINER_REGISTRY_TOKEN}" -Djib.to.tags="${IMAGE_TAG}"
```
[the remark this answers is not in the document]
Comment (horozal) [reply .github/workflows/publish-container-image.yml]: Removed `-pl smartclide-monitoring -Djib.to.auth.username="${CONTAINER_REGISTRY_USERNAME}" -Djib.to.auth.password="${CONTAINER_REGISTRY_TOKEN}"`

File: pom.xml [hunk 6]
```diff
@@ -56,7 +57,40 @@
...
 				<artifactId>spring-boot-maven-plugin</artifactId>
 			</plugin>
+			
+			<plugin>
+				<groupId>com.google.cloud.tools</groupId>
+				<artifactId>jib-maven-plugin</artifactId>
+				<version>${jib-maven-plugin.version}</version>
+				<configuration>
+				    <from>
+					<image>openjdk:11.0.10-jre-slim-buster</image>
```
[the remark this answers is not in the document]
Comment (horozal) [reply pom.xml]: OK, replaced `openjdk:11.0.10-jre-slim-buster` with `eclipse-temurin:11-jre-jammy`

File: pom.xml [hunk 7]
```diff
@@ -56,7 +57,40 @@
...
+				    </from>
+				    <to>
+					<image>ghcr.io/eclipse-opensmartclide/smartclide/${project.artifactId}:latest</image>
+				    </to>
+				    <extraDirectories>
```
[the remark this answers is not in the document]
Comment (horozal) [reply pom.xml]: Removed the script and its specification in the pom file.

**Review by schlotze — APPROVED** [review]

**Pull request merged**